### PR TITLE
use spectral width for x data in nmr jcamp generation

### LINKED ISSUE #64: ensure descending values in x scale for nmr jcamp generation
currently, the data is passed to the encoder without taking care of the order, seems it should be descend to ensure a correct x axis generation because the value of the LDR `OFFSET` is the value of the first element of the x axis.

### PR DESCRIPTION
**Pull request by jobo322**
reduce size of jcamp using the spectral width and the number of points it is similar to the exported jcamp from bruker topspin

### CODE REVIEW

File: src/from1DNMRVariables.ts [hunk 1]
```diff
@@ -17,13 +19,17 @@ type NtuplesData<DataType extends DoubleArray = DoubleArray> = Record<
...
 type PeakData<DataType extends DoubleArray = DoubleArray> = Record<
   'x' | 'y',
   MeasurementVariable<DataType>
 >;
 
 const peakDataKeys = ['x', 'y'] as Array<keyof PeakData>;
-const ntuplesKeys = ['x', 'r', 'i'] as Array<keyof NtuplesData>;
+const ntuplesKeys = ['r', 'i'] as Array<keyof NtuplesData>;
 
 function isPeakData(
```
Comment (lpatiny) [inline src/from1DNMRVariables.ts]: It should not be called PeakData but XYData (please change everywhere). PeakData is used for mass spectrometry fragmentation but for NMR we have equidistant x values.

File: src/from1DNMRVariables.ts [hunk 2]
```diff
@@ -133,11 +169,11 @@ export function from1DNMRVariables(
 ##ORIGIN=${origin}
 ##OWNER=${owner}\n`;
 
-  const infoKeys = Object.keys(info).filter(
+  const infoKeys = Object.keys(newInfo).filter(
     (e) =>
```
Comment (lpatiny) [inline src/from1DNMRVariables.ts]: Please rename `e` to `key`

File: src/from1DNMRVariables.ts [hunk 3]
```diff
@@ -185,3 +221,21 @@ export function from1DNMRVariables(
...
 }
+
+function checkDescending(variables: NMR1DVariables) {
+  const xVariable = variables.x as MeasurementVariable<DoubleArray>;
+  const reverse =
+    xVariable.units !== 'time' && xVariable.data[0] < xVariable.data[1];
+
+  if (!reverse) return variables;
+
+  const newVariables = JSON.parse(
```
Comment (lpatiny) [inline src/from1DNMRVariables.ts]: This is extremely slow and should not be used. We need to make a zoom to discuss about this PR.

**Review by lpatiny — CHANGES_REQUESTED** [review]

**Comment by targos** [comment]
No tests?

**Pull request merged**